packages/studioCMSBlog/src/components/PageList.astro

```diff
@@ -0,0 +1,118 @@
+---
+import { CImage, FormattedDate, type pageDataReponse } from 'studiocms:components';
+
+interface Props {
+    blogPageList: pageDataReponse[];
+}
+
+const { blogPageList } = Astro.props;
+---
+
+<ul>
+    { 
+       blogPageList.length > 0 && blogPageList.map(({slug, heroImage, title, description, publishedAt}) => (
+           <li>
+               <a href={`${import.meta.env.BASE_URL}blog/${slug}/`}>
```

Comment:
This interpolation is not safe

Also, it is possible for users to change the path `/blog/[...slug].astro`. The final path of the slug should be retrieved using the context module:

```tsx
import { pages } from 'studiocms-blog:context';

const blogPath = pages
	.get('/blog/[...slug].astro')
	// trim dynamic slug from end
	.replace(/\/[^]*$/, '')

if (blogPath) {
	// ...
} 
```

Reply:
> Also, it is possible for users to change the path /blog/[...slug].astro....
erm.... that is something im not sure how i'm going to fix.... since i actually define the Blog index route statically above outside of the actual integration for the Maps...

```ts
externalNavigation.set("@astrolicious/studiocms-blog/index", {text: "Blog", slug: "blog/"});
```

is there a good way to include these within the actual ATP logic?  ATP is still a strange strange beast to me...

Reply:
I am not too sure, `externalNavigation.set(...)` runs at buildtime, but the virtual module ATP creates to expose this data is only available at runtime 🤔 I guess ATP would need a way of exposing this data at buildtime, but I am not exactly sure what the API would look like for that.

Reply:
Welp.  Thats now the only thing that would break if a user changes that config option... lol the actual path routing will work now.... but just that one link that gets called from that function.... :rofl: